config.py

```diff
@@ -100,6 +104,51 @@ def validate_config(config):
             ):
                 raise ConfigError("Config error: Ignored tables parameter should be a list")
 
+    if "notification" in config:
+        settings = [
+            "smtp_server",
+            "email_sender",
+            "email_subject",
+            "email_recipients",
+        ]
+
+        for setting in settings:
+            if setting not in config.notification:
+                raise ConfigError(f"Config error: `{setting}` is missing from `notification`.")
+
+        if not isinstance(config.notification.email_recipients, list):
+            raise ConfigError("Config error: `email_recipients` should be list of addresses.")
+
+        if "use_ssl" in config.notification:
+            if not isinstance(config.notification.use_ssl, bool):
+                raise ConfigError("Config error: `use_ssl` must be set to either `true` or `false`.")
+
+        if "use_tls" in config.notification:
+            if not isinstance(config.notification.use_tls, bool):
+                raise ConfigError("Config error: `use_tls` must be set to either `true` or `false`.")
+
+        if "smtp_port" in config.notification:
+            if not isinstance(config.notification.smtp_port, int):
+                raise ConfigError("Config error: `smtp_port` must be set an integer.")
+
+        if "minimal_email_interval" in config.notification:
+            if not isinstance(config.notification.minimal_email_interval, (int, float)):
+                raise ConfigError("Config error: `minimal_email_interval` must be set a number.")
+
+        smtp_conn: smtplib.SMTP = None
+
+        try:
+            smtp_conn = create_connection(config)
+        except OSError:
```

Comment:
Documentation says connect raises SMTPConnectError... and it would be good to also report the exception's error message to help debugging

Reply:
You can also have `socket.gaierror` if the server does not exist at all. So I capture and resend the message for all the errors that may occur.